### ISSUE TITLE
Loss of Application Icon for Pinning to Dash After Upgrade on Arch Linux

### ISSUE THREAD

**Issue by C0sm0cats** [wavetermdev/waveterm#863]
when I upgrade from version 0.7.6-1 to 0.8.5-1 on Arch Linux I lose the application icon to pin to the Dash.

**Comment by esimkowitz** [comment]
Yeah this is likely because we renamed the Linux binary, are you able to re-add it?

**Comment by C0sm0cats** [comment]
Hello. I’m not sure how to re-enable pinning waveterm to the dash-to-panel. Do you have any suggestions on how to restore that functionality?

Additionally, I previously uninstalled version 0.7.6-1 and installed the pacman version 0.8.6-1. When I run it from the command line, I receive the following error message:

"shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
chdir: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
error configuring updater Error: ENOENT: no such file or directory, open '/usr/lib/electron32/resources/app-update.yml"

WR

**Comment by C0sm0cats** [comment]
Hello,
I wanted to provide an update regarding the installation of Waveterm. Here are the key details:

 The binary is located at /usr/bin/waveterm
-rw-r-xr-x root root 1.1 KB Fri Sep 27 08:06:49 2024 waveterm

The .desktop file is located at /usr/share/applications/waveterm.desktop
    -rw-r--r-- root root 292 B Fri Sep 27 08:06:49 2024 waveterm.desktop

Modification Summary
I made the following modification to the waveterm.desktop file to ensure it points to the correct executable:
Original line: Exec=/opt/Wave/waveterm %U
Updated line: Exec=/usr/bin/waveterm %U

This change resolved the issue, and Waveterm is now successfully pinned to the Dash to Panel with the correct functionality.

**Comment by esimkowitz** [comment]
Thank you for this, sounds like we need a custom postinstall script

**Comment by C0sm0cats** [comment]
You're welcome! And absolutely, a custom postinstall script would be ideal for this scenario. By the way, I must say that Waveterm is undoubtedly the best terminal editor out there right now and works perfectly on Manjaro. congratulations on such an outstanding tool!

**Comment by C0sm0cats** [comment]
It would be amazing if AI could analyze the code in a separate Waveterm block and assist with development, even though it seems quite challenging. It's something I dream of!

**Comment by esimkowitz** [comment]
Yes!! We're really excited about the AI functionality. We're experimenting now with giving it more context into what's actually happening in the app so it can assist you better

**Comment by esimkowitz** [comment]
Hey @C0sm0cats I'm a bit confused as to why the desktop needs to point to /usr/bin/waveterm, I am not seeing the same issue on my computer, despite the desktop entry pointing to /opt/Wave/waveterm.

Did you have an old desktop entry for the v0.7 version of the app?

**Comment by C0sm0cats** [comment]
Hi [@esimkowitz ](https://github.com/esimkowitz) Yes, I did have an older version installed, which was installed using the command: sudo pacman -U Wave-linux-x64-0.7.6.pacman

**Comment by esimkowitz** [comment]
Closing this as unplanned
